tests/test-lxd-rbac

```diff
@@ -0,0 +1,189 @@
+#!/bin/sh
+set -eu
+
+waitSnapdSeed() (
+  set +x
+  for i in $(seq 60); do # Wait up to 60s.
+    if systemctl show snapd.seeded.service --value --property SubState | grep -qx exited; then
+      return 0 # Success.
+    fi
+
+    sleep 1
+  done
+
+  echo "snapd not seeded after ${i}s"
+  return 1 # Failed.
+)
+
+cleanup() {
+    echo ""
+    if [ "${FAIL}" = "1" ]; then
+        echo "Test failed"
+        exit 1
+    fi
+
+    echo "Test passed"
+    exit 0
+}
+
+FAIL=1
+trap cleanup EXIT HUP INT TERM
+
+# Wait for snapd seeding
+waitSnapdSeed
+
+# Install LXD
+while [ -e /usr/bin/lxd ]; do
+    apt-get remove --purge --yes lxd lxd-client lxcfs liblxc1
+done
+apt-get remove --purge cloud-init --yes
+snap remove lxd || true
+snap install lxd --channel=latest/edge
+apt-get install --no-install-recommends --yes jq
+snap install bhttp
+lxd waitready --timeout=300
+
+# Configure LXD
+lxc storage create default zfs
+lxc profile device add default root disk path=/ pool=default
+lxc network create lxdbr0
+lxc profile device add default eth0 nic network=lxdbr0 name=eth0
+
+# Configure RBAC
+curl https://canonical-lxd.monstermunchkin.dev/config/rbac-test.service -o rbac-test.service
```

Comment:
I don't know if the private keys from that file should be openly accessible? If they belong to a "demo" key, I'd rather replace the `curl | jq` by hardcoding them in the `lxc config set`.

The idea is to avoid depending on an external source.

Reply:
I'm hoping that @monstermunchkin will be able to figure out how to automate the setup of the rbac server so we can spin one up temporarily as part of this test, as we won't be able to depend on his server for running the automated tests.